src/main/java/org/myrobotlab/service/ProgramAB.java

```diff
@@ -1427,8 +1339,8 @@ public void onUtterance(Utterance utterance) throws Exception {
         // TODO: don't talk to bots.. it won't go well..
         // TODO: the discord api can provide use the list of mentioned users.
         // for now.. we'll just see if we see Mr. Turing as a substring.
-        sleep = (sleep || utterance.text.contains("@")) && !utterance.text.contains(botName);
-        if (!sleep) {
+        c.sleep = (c.sleep || utterance.text.contains("@")) && !utterance.text.contains(botName);

```

Comment:
If I understand this correctly, I think it would cause the bot to wake up if its name is mentioned anywhere in the utterance. Could be problematic if people are talking about the bot but not directly talking to it.

Also, just searching for the @ symbol makes it impossible to include it in a message to the bot, so things like email addresses would cause it to go to sleep.

Could there be a `sleepListener` interface instead, that Discord could use to set the sleep state based on its own conditions? I think this would be useful for other services too, like Inmoov so other services can set the sleep state and listen for the sleep state changes

Reply:
Yeah, sounds like a good idea.  "Utterance" vs plain text is supposed to keep some of the necessary meta data in order to do appropriate processing .. vs just a text in a string.  I agree it seems very Discord centric and the loss of talking about email.  This pr is only about normalizing the member variables with config, but a SleepListener sounds like a good idea for a future one :+1: